### ISSUE TITLE
Flat config with ESM modules doesn’t work

### ISSUE THREAD

**Issue by cherryblossom000** [eslint-functional/eslint-plugin-functional#791]
## Bug Report

Rules such as `functional/type-declaration-immutability` don’t work when using the flat config with ESM modules.

`test.ts`:
```ts
type Test = readonly string[]
```

`eslint.config.js`:
```js
import tseslint from 'typescript-eslint'
import functional from 'eslint-plugin-functional/flat'

export default tseslint.config(
	{
		files: ['**/*.ts'],
		languageOptions: {
			ecmaVersion: 2022,
			parserOptions: {
				project: true,
				tsconfigRootDir: import.meta.dirname,
			},
		},
	},
	tseslint.configs.base,
	functional.configs.strict,
	{
		rules: {
			'functional/type-declaration-immutability': 'error',
		},
	},
)
```

### Expected behavior

No errors.

### Actual behavior

```
/path/to/file/test.ts
  1:1  error  This type is declare to have an immutability of at least "Immutable" (actual: "Unknown")  functional/type-declaration-immutability
```

### Steps to reproduce

1. Install eslint-plugin-function@6.2.0-next.3
2. Run `eslint test.ts`

### Proposed changes

https://github.com/eslint-functional/eslint-plugin-functional/blob/72c6f67b43bb19dbb88c98d408c60b63cb0f088d/src/utils/conditional-imports/typescript.ts#L3-L9

`typescript` can’t be imported in an ESM module as `require` is not defined.
